src/detect-smb-fp.c

```diff
@@ -0,0 +1,381 @@
+/* Copyright (C) 2022 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#include "suricata-common.h"
+
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-smb-fp.h"
+
+#include "rust.h"
+
+static void DetectSMBfingerprintFree(DetectEngineCtx *de_ctx, void *ptr)
+{
+    SCFree(ptr);
+}
+
+typedef struct DetectSMB_Fingerprint {
+    char md5[SC_MD5_LEN];
+} DetectSMB_Fingerprint;
+
+const uint8_t SC_hexvalue[256] = {
```

Comment:
formatting is weird... if this is clang-format maybe we should manually format and disable clang-format for this block

Reply:
Yes, it was clang-format.
I had 16 lines of 16 values IIRC

> maybe we should manually format and disable clang-format for this block

How can I do so ?

Reply:
e.g. https://github.com/OISF/suricata/blob/master/src/suricata.c#L1296